design/pool-configuration.md

```diff
@@ -0,0 +1,552 @@
+* using the `auto-assign: false` parameter on the pool, and then statically assign the IP to a given service.
+
+From this point of view, we can enhance the allocation logic allowing users to reserve a given pool for a set of namespaces or for a set
+of services. This is more flexible and moves the focus on the cluster administrator, who is setting the MetalLB configuration
+in a multi tenant scenario.
+
+This, in combination with the other configuration degrees above, will provide a way to have per service (or per namespace) behavior
+ (i.e. specifying the communities for a given service).
+
+The issues related to this arguments:
+
+* [Issue: Pool affinity for Namespace](https://github.com/metallb/metallb/issues/383)
+* [Issue: Feature request: addresses assignment outside of Service definition](https://github.com/metallb/metallb/issues/616)
+* [Issue: allow selecting services to announce with the bgb speaker](https://github.com/metallb/metallb/issues/604)
+* [Issue: Dynamic BGP communities from services](https://github.com/metallb/metallb/issues/934)
+
+### Risks and Mitigations
+
+#### Regressions
+
+The e2e test suite is wide enough to ensure we have good coverage on metallb's main features. Given the new data model will be functionally equivalent
+to the old one, converting the test suite should give us enough confidence in that regard.
+
+#### Impacts on the users' configurations
+
+Deprecating the configmap will have impacts on the users' current configurations. To mitigate this, we may consider having a controller that converts the configmap
+into the CRDs, or provide some offline tool in charge of that (the latter option is more desirable, as it will remove the need of reconciling changes into the configmap).
+
+#### Impacts on users currently using the MetalLB operator
+
+The operator already exposes CRDs, mapped directly to the current configmap entitites.
+Changing the CRDs layout will impact them, and in order to mitigate the behaviour, conversion webhooks will be implemented to support both the old and the new versions
+of the API, until the old version is deprecated.
+
+#### Bad configurations
+
+##### Service allocation
+
+When a user specify both a pool to be allocated to a specific service and one with no selector, the service may receive an IP from the broader one.
+We need to implement an easy to understand and deterministic behaviour (see design details below).
+
+##### Non overlapping filters
+
+Having an IP with no nodes advertising it is a risk. Some scenarios are:
+
+* Traffic Policy Local with pods not running on the same nodes selected by the selector
+* Node BGP peer selector with no intersection with the address selector
+
+It should be possible to detect non overlapping configurations, and raise an event when they happen. The metallb operator introduces a Validation WebHook
+that can be used to prevent those configurations to be created in first instance.
```

Comment:
```suggestion
that can be used to prevent those configurations from being created in first instance.
```

Reply:
>in first instance

Did you mean "prevent from the start" or "prevent the 1st use case" (from the 2 cases mentioned above)?